app/controllers/metrics_controller.rb

```diff
@@ -1,18 +1,18 @@
 class MetricsController < ApplicationController
   before_action :authorize_prison_request
+  before_action :authenticate_user, only: :send_confirmed_bookings
 
   def index
     @prisons = PrisonsDecorator.decorate(Prison.enabled.includes(:estate))
     @dataset = MetricsPresenter.new(all_time_counts)
     @graphs_presenter = GraphMetricsPresenter.new
   end
 
-  def confirmed_bookings
-    exporter = WeeklyMetricsConfirmedCsvExporter.new(weeks: 12)
+  def send_confirmed_bookings
```

Comment:
Personally not a huge fan of none standard actions in controllers as it's not what you would be expecting in a controller. Shouldn't be seeing any other public methods than `index, show, create, update, delete` in a controller.
One of the main benefits is that it prevents a long list of semantically incorrect `get,put,post,member,collection` macros in the routes file which quickly gets out of hand and typically does not get any love from developers.

The usual pattern is create a controller `SendConfirmedBookingsController` with a `#create` action a `resources :send_confirmed_bookings, only: create`.

Reply:
If this was a RESTful API it would make sense, I see little benefit in this case